### ISSUE TITLE
Feat: Add Undo and Redo Buttons in GridPaint Activity

### ISSUE THREAD

**Issue by Devmoni** [llaske/sugarizer#1694]
### Description

Currently, the activity lacks the ability to undo or redo actions, which can be tough for users who make mistakes or want to revert changes. Adding undo and redo buttons will enhance the user experience by providing more control over the drawing process.

**Comment by Devmoni** [comment]
I have already implemented this and am waiting for the verification of the issue. 
Please review @llaske 
Thanks...

**Comment by llaske** [comment]
GridPaint is the most used activity by young children.
I don't want to add complexity in the UX. Plus, it's very easy to redraw something.
So I don't think we need an undo/redo feature.

**Comment by Devmoni** [comment]
I respect your decision.
 However, considering that the ```clear-button``` removes all painted blocks, having an undo/redo feature could be very useful for correcting small mistakes without needing to start over. 
This functionality is already available in various activities on Sugarizer.

**Comment by Devmoni** [comment]
> GridPaint is the most used activity by young children. I don't want to add complexity in the UX. Plus, it's very easy to redraw something. So I don't think we need an undo/redo feature.

I agree that one extra button will get added in the UI but IMO its worth adding one button somewhere it will increase the ease to draw something. 
Moreover nowadays undo/redo feature is used everywhere teaching young children how to use it from an early age will certainly be beneficial.
Thanks...
